app/src/androidTest/java/org/kiwix/kiwixmobile/testutils/TestUtils.java

```diff
@@ -18,11 +18,16 @@
 package org.kiwix.kiwixmobile.testutils;
```

Comment:
These changes are quite extensive and rely on digging deep into Espresso and monitoring Android activities in order for the code to do what we want it to do. How about adding a comment block with the goals and challenges you're trying to resolve and the approach you've taken. This may enable future contributors to distinguish between the intent and the current implementation if and when this code breaks.

Reply:
I've changed much of the code that was here when you wrote this comment - do you think this still applies?